subprojects/docs/src/docs/userguide/authoring-builds/introduction/writing_build_scripts.adoc

```diff
@@ -0,0 +1,559 @@
+The locations of the binaries for the dependencies must be described.
+
+[source]
+----
+repositories {
+    mavenCentral()
+}
+----
+
+In the example, the `guava` library and the JetBrains kotlin plugin (`org.jetbrains.kotlin.jvm`) will be downloaded from the link:https://repo.maven.apache.org/maven2/[Maven Central Repository].
+
+=== Add dependencies
+
+A project generally has a number of dependencies it needs in order to do its work.
+These dependencies are typically imports in the source code.
+
+Dependencies are grouped in configurations, and can be retrieved and uploaded from repositories.
+
+Use the `DependencyHandler` returned by `Project.getDependencies()` method to manage the dependencies.
+Use the `RepositoryHandler` returned by `Project.getRepositories()` method to manage the repositories.
+
+[source]
+----
+dependencies {
+    testImplementation("org.jetbrains.kotlin:kotlin-test-junit5")
+    testImplementation("org.junit.jupiter:junit-jupiter-engine:5.9.3")
+    testRuntimeOnly("org.junit.platform:junit-platform-launcher")
+    implementation("com.google.guava:guava:32.1.1-jre")
+}
+----
+
+In the example, the application code uses the `guava` libraries from Google.
+Guava provides utility methods for collections, caching, primitives support, concurrency, common annotations, string processing, I/O, and validations.
+
+=== Use convention properties
+
+A plugin can add properties and method to a project through the project's `Convention` object.
+
+In the example, the `application` plugin added an `application` property which is used to detail the main class of our Java application:
+
+[source]
+----
+application {
+    mainClass.set("main")
+}
+----
+
+=== Register and configure tasks
+
+Tasks performs some basic piece of work, such as compiling classes, or running unit tests, or zipping up a WAR file.
+
+While tasks are typically defined in plugins, you may need to register or configure tasks in build scripts.
+
+*Registering* a task adds the task to your project.
+
+You can register tasks in a project using the link:{javadocPath}/org/gradle/api/tasks/TaskContainer.html[`TaskContainer.register(java.lang.String)`] method:
+
+[source]
+----
+tasks.register<Zip>("zip-reports") {
+    from 'Reports/'
+    include '*'
+    archiveName 'Reports.zip'
+    destinationDir(file('/dir'))
+}
+----
+
+You may have seen usage of the link:{javadocPath}/org/gradle/api/tasks/TaskContainer.html[`TaskContainer.create(java.lang.String)`] method *which should be avoided*:
+
+[source]
+----
+tasks.create<Zip>("zip-reports") {
+    from 'Reports/'
+    include '*'
+    archiveName 'Reports.zip'
+    destinationDir(file('/dir'))
+}
+----
+
+TIP: `register()`, which enables <<task_configuration_avoidance.adoc#task_configuration_avoidance,task configuration avoidance>>, is preferred over `create()`.
+
+You can acquire a reference to a task to configure it using the `TaskCollection.named(java.lang.String)` method:
```

Comment:
💅 This is true, but you're not explicitly storing the reference anywhere in a variable.  Maybe just "locate" a task to configure it?